lib/galaxy/jobs/runners/arc.py

```diff
@@ -0,0 +1,593 @@
+                """ App tag """
+                sub_el = SubElement(app_exe,"Path")
+                sub_el.text = "./" + dataset_name
+
+
+        """ Populate datastaging output tag with all output files - in addition to populate the arcjob object"""
+        """ Potentially more than one file - but currently actually only one, so the for-loop here is currently not actually needed """
+ 
+        """ Fill the appropriate job description fields """
+        sub_el = SubElement(datastaging,"OutputFile")
+        subsub_el = SubElement(sub_el,"Name")
+        subsub_el.text = "/"
+        
+
+        """ This is hardcoded stdout and stderr files from the ARC job defined here in the runner - TODO - not hardcoded """
+        sub_el = SubElement(datastaging,"OutputFile")
+        subsub_el = SubElement(sub_el,"Name")
+        subsub_el.text = std_out
+
+        """ This is hardcoded stdout and stderr files from the ARC job defined here in the runner - TODO - not hardcoded """
+        sub_el = SubElement(datastaging,"OutputFile")
+        subsub_el = SubElement(sub_el,"Name")
+        subsub_el.text = std_err
+
+        """ TODO - just a sample, this will probably be set by the destination itself - to be discussed """
+        cpuhrs = arc_cpuhrs
+        sub_el = SubElement(resources,"IndividualCPUTime")
+        sub_el.text = arc_cpuhrs
+
+        """ TODO - just a sample, this will probably be set by the destination itself - to be discussed """
+        mem = arc_mem
+        sub_el = SubElement(resources,"IndividualPhysicalMemory")
+        sub_el.text = arc_mem
+
+        """ Populate the arcjob object with rest of necessary and useful fields including the full job description string"""
+        """ All files that should be collected by ARC when the job is finished need to be appended to the downloadFiles list - 
+        here it is just the folder / and all files in the folder will be downloaded. 
+        The arc.py in pyarcrest loops over this list to fetch all outputfiles  """
+        arcjob.downloadFiles.append("/")
+        arcjob.StdOut = std_out
+        arcjob.StdErr = std_err
+        arcjob.RequestedTotalCPUTime = cpuhrs
+        arcjob.descstr = tostring(descr, encoding='unicode',method='xml')
+
+
+    def job_actions(self, arcid, action):
+
+
+        """ Kill job in ARC.
+            Pass job_id to http_delete_request method.
+        """
+        """ Really processing one job at a time, should I do this better? pyarcrest/arc.py takes a list of jobs"""
+
+        if action == "status":
+            results = self.arcrest.getJobsStatus([arcid])
+            for state in results:
+                try:
+                    job_state = state
+                    return  job_state
+                except Exception:
+                    log.error(f'Could not get status of job with id: {arcid}')
+        elif action == "kill":
+            results = self.arcrest.killJobs([arcid])
+            for killed in results:
```

Comment:
This will return after the first iteration, so really there's no iteration - is that intentional?
Also, you could just `return bool(killed)` instead of the if/then.